### PR TITLE
[DOCS] Ensuring consistent style of directories, files, and related references in docs/

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-  GDOC-126: https://superconductive.atlassian.net/browse/GDOC-126?atlOrigin=eyJpIjoiYTM5MTRlYThkMjlkNDIwOTg2MzVmMWRmN2ExYWE0YzMiLCJwIjoiaiJ9
- Change of directories and references from "-" to "_"
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Review by cdkini — COMMENTED** [review]
I made these changes using a combination of `grep` and `sed`. I've done a review of the changes herein but please do take a detailed look to ensure that nothing has changed erroneously.

The fact that the Netlify build is working is probably a sign that things are changed as intended.

File: docs/deployment_patterns/how-to-instantiate-a-data-context-hosted-environments.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Step 1: Configure your Data Context
 
 The following guide gives an overview of creating an in-code Data Context including defaults to help you more quickly set one up for common configurations:
 
-- [How to instantiate a DataContext without a YML file](../guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file)
+- [How to instantiate a DataContext without a YML file](../guides/setup/configuring_data_contexts/how-to-instantiate-a-data-context-without-a-yml-file)
```
Comment (alexsherstinsky) [inline docs/deployment_patterns/how-to-instantiate-a-data-context-hosted-environments.md]: @cdkini GitHub still shows me file names containing hyphens, while the rest of the path uses the underscores.  Thanks.

File: docs/deployment_patterns/how-to-instantiate-a-data-context-hosted-environments.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ In order to use an Expectation Suite you've created to validate data, follow thi
 Step 4: Use Data Docs
 ----------------------
 
-Finally, if you would like to build and view Data Docs in your environment, please follow the guides for configuring Data Docs: [Options for hosting Data Docs](../tutorials/getting-started/customize-your-deployment#options-for-hosting-data-docs).
+Finally, if you would like to build and view Data Docs in your environment, please follow the guides for configuring Data Docs: [Options for hosting Data Docs](../tutorials/getting_started/customize-your-deployment#options-for-hosting-data-docs).
```
Comment (alexsherstinsky) [inline docs/deployment_patterns/how-to-instantiate-a-data-context-hosted-environments.md]: @cdkini GitHub still shows me file names containing hyphens, while the rest of the path uses the underscores.  Thanks.

File: docs/deployment_patterns/how-to-instantiate-a-data-context-on-an-emr-spark-cluster.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ Steps
 
 2. **Configure a Data Context in code.**
 
-    Follow the steps for creating an in-code Data Context in [How to instantiate a Data Context without a yml file](../guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file.md).
+    Follow the steps for creating an in-code Data Context in [How to instantiate a Data Context without a yml file](../guides/setup/configuring_data_contexts/how-to-instantiate-a-data-context-without-a-yml-file.md).
```
Comment (alexsherstinsky) [inline docs/deployment_patterns/how-to-instantiate-a-data-context-on-an-emr-spark-cluster.md]: @cdkini GitHub still shows me file names containing hyphens, while the rest of the path uses the underscores.  Thanks.

File: docs/deployment_patterns/how-to-instantiate-a-data-context-on-databricks-spark-cluster.md [hunk 4]
```diff
@@ -5,11 +5,11 @@ import Prerequisites from '../guides/connecting_to_your_data/components/prerequi
 
 This guide will help you instantiate a Data Context on an Databricks Spark cluster.
 
-The guide demonstrates the recommended path for instantiating a Data Context without a full configuration directory and without using the Great Expectations [command line interface (CLI)](../guides/setup/configuring-data-contexts/how-to-create-a-new-data-context-with-the-cli.md).
+The guide demonstrates the recommended path for instantiating a Data Context without a full configuration directory and without using the Great Expectations [command line interface (CLI)](../guides/setup/configuring_data_contexts/how-to-create-a-new-data-context-with-the-cli.md).
```
Comment (alexsherstinsky) [inline docs/deployment_patterns/how-to-instantiate-a-data-context-on-databricks-spark-cluster.md]: @cdkini GitHub still shows me file names containing hyphens, while the rest of the path uses the underscores.  Thanks.

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@cdkini GitHub still shows me file names containing hyphens, while the rest of the path uses the underscores.  What was decided in terms of the standardization of the names of the documentation files?  Thanks!

**Comment by cdkini** [comment]
Script used to perform file changes (`fd` is an updated `find` and `sd` is a simplified `sed`)

```
#!/bin/bash

FILES=$(fd .)

for FILE in ${FILES}; do
    CHANGE=$(echo $FILE | tr "-" "_")
    git mv $FILE $CHANGE
    rg $FILE -l | xargs sd $FILE $CHANGE
done
```

**Review by cdkini — COMMENTED** [review]
Did a manual review here but would appreciate additional eyes. My implementation of `grep` and `sed` wasn't as clean as I would've liked and I had to go back to manual change a few things.

File: docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md [hunk 5]
```diff
@@ -61,15 +61,15 @@ Steps
 2. If using a YAML file, set the ``config_variables_file_path`` key in your ``great_expectations.yml`` or leave the default.
 
   ```yaml
-  # great_expectations/great_expectations.yml
+  # great-expectations/great_expectations.yml
```
Comment (alexsherstinsky) [inline docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md]: This does not seem to be correct -- should this not remain using the underscore?  If agree, then please verify all similar cases (and must be different from the location on GitHub where the hyphen is used).

File: docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md [hunk 6]
```diff
@@ -61,15 +61,15 @@ Steps
...
 
   config_variables_file_path: uncommitted/config_variables.yml
   ```
 
 3. Replace credentials or other values in your ``great_expectations.yml`` with ``${}``-wrapped variable names (i.e. ``${ENVIRONMENT_VARIABLE}`` or ``${YAML_KEY}``).
 
   ```yaml
-  # great_expectations/great_expectations.yml
+  # great-expectations/great_expectations.yml
```
Comment (alexsherstinsky) [inline docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md]: This does not seem to be correct -- should this not remain using the underscore?  If agree, then please verify all similar cases (and must be different from the location on GitHub where the hyphen is used).

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@cdkini I found one potential issue -- please see the comment(s).  Thanks!

File: docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md (hunk 5 above)
Comment (cdkini) [reply docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md]: Ah yes I erroneously changed all instances to underscores and then went back to revise anything that linked to the GE repo. Forgot to account for this let me change it real quick.

File: docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md (hunk 6 above)
Comment (cdkini) [reply docs/guides/setup/configuring_data_contexts/how_to_configure_credentials_using_a_yaml_file_or_environment_variables.md]: Changed this as well. Great catch, thank you so much!

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**